yarn-project/archiver/src/archiver/data_retrieval.ts

```diff
@@ -103,6 +103,10 @@ export async function retrieveBlocksFromRollup(
   logger: Logger = createLogger('archiver'),
 ): Promise<RetrievedL2Block[]> {
   const retrievedBlocks: RetrievedL2Block[] = [];
+
+  let chainId: Fr | undefined;
+  let version: Fr | undefined;
```

Comment:
```suggestion
  const [chainId, version] = await Promise.all([
      new Fr(publicClient.getChainId()),
      new Fr(await rollup.read.getVersion())
  ]);
```

We can just fetch them here and not deal with the conditionals below?